pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -341,6 +341,80 @@ func (c *Cluster) checkHealth(ctx context.Context) error {
 	return nil
 }
 
+// determineExtraMonToRemove assumes all mons are in quorum and that there are more mons
+// that required for desired state. One mon will be picked for removal in this priority:
+// 1. If a stretch cluster, remove the extra mon according to the stretch topology
+// 2. If more than one mon on a node, remove one of them
+// 3. If no criteria require for 1 or 2, pick an arbitrary mon
+func (c *Cluster) determineExtraMonToRemove() string {
+	mons := c.clusterInfoToMonConfig()
+	if c.spec.IsStretchCluster() {
+		stretchMonToRemove := c.findExtraMonToRemoveFromStretchCluster(mons)
+		if stretchMonToRemove != "" {
+			return stretchMonToRemove
+		}
+		logger.Infof("did not find an extra mon to remove from the stretch cluster")
+		return ""
+	}
+
+	nodesWithMons := map[string]string{}
+	arbitraryMon := ""
+	for _, mon := range mons {
+		if mon.NodeName == "" {
+			logger.Infof("mon %q is not assigned a host yet", mon.DaemonName)
```

Comment:
Two questions.

a. IIUC, if `c.spec.Network.IsHost() || c.monVolumeClaimTemplate(mon) == nil` is `false`, mon pod scheduling is done by K8s scheduler\[1\] and `mon.NodeName` become "" for such mons (this assignment is introduced in this PR). If so, isn't this log improper in such case?
b. `determineExtraMonToRemove` assumes "all mons are in quorum" as described in the function's comment. Then shouldn't all mons have NodeName?

\[1\]: https://github.com/rook/rook/blob/cc15c94e6c4e64fb424dd1aa95d09fb3928dff15/pkg/operator/ceph/cluster/mon/mon.go#L867-L893